### PR TITLE
Sustain tk mehzyg improve code revision check

### PR DESCRIPTION
**Pull request by pblesi**
It would be nice if litmus paper's `Dependency::Script` could additionally report its output in Litmus's status. The intended use case for this is to have Litmus report the current revision of code that is running. 

This PR adds a `report_result` option to Litmus's `Dependency::Script` dependency that will enable the dependency to return the output from the script and have it printed in Litmus's status.

Note: Add `?w=1` for a cleaner diff

Concerns:
I added a constraint to only return 100 characters of output from the script to litmus status. I did this to prevent scripts from spamming the returned status. 100 was chosen arbitrarily. Should this be changed? Is it necessary?

### CODE REVIEW

File: lib/litmus_paper/dependency/base.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module LitmusPaper
+  module Dependency
+    class Base
+      attr_reader :available
+
+      def available?
+        return @available unless @available.nil?
```
Comment (dkuntz2) [inline lib/litmus_paper/dependency/base.rb]: This looks like it's caching the result, which seems undesirable in basically all cases. Unless I'm reading this wrong?

Comment (dkuntz2) [inline lib/litmus_paper/dependency/base.rb]: So doing some research, based on where litmus actually creates the dependency objects ([in service.rb](https://github.com/braintree/litmus_paper/blob/c237b4d82f3eda6ad0867edcc8c351f07b6e694c/lib/litmus_paper/service.rb#L27-L29)), and that services are a part of the litmus config, it just calls `dependency.available?` each time a request is made, without creating new objects.

Comment (pblesi) [inline lib/litmus_paper/dependency/base.rb]: @dkuntz2 You're right. I was thinking the dependency objects were reinitialized every health check, but it looks like they exist for the lifetime of the process. I added 2 commits to fix this.

**Comment by dkuntz2** [comment]
LGTM

**Comment by pblesi** [comment]
I'm going to close this PR as we found a more straight-forward solution to report the deployed code revision.

**Pull request closed without merging**